File: C++/22.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        '''
        :type: prices: vector<int>
        rtype: int
        '''
        int profit = 0;
        for (int i = 1; i < prices.size(); i++) {
            if (prices[i] > prices[i-1]) 
                profit += (prices[i] - prices[i-1]);
        }
        return profit;
    }
};

File: Python/22.py
class Solution:
    def generateParenthesis(self, n: int) -> List[str]:
        '''
        :type: n: int
        :rtype: List[str]
        '''
        res = []
        self.dfs(n, 1, 0, '(', res)
        return res
        
    def dfs(self, n, l_num, r_num, path, res):
        if l_num == n:
            if r_num == n:
                res.append(path)
                return
            else:
                self.dfs(n, l_num, r_num+1, path+')', res)
        else: # if l_num < n:
            self.dfs(n, l_num+1, r_num, path+'(', res)
            if l_num > r_num:
                self.dfs(n, l_num, r_num+1, path+')', res)